wordpress-mobile/WordPress-iOS
# Replaces the different checks for jetpack branding visibility with a single flag

Fixes #NA

This PR introduces `JetpackBrandingVisibility`, an `enum` to set the visibility of the jetpack branding ui components (badges and banners). The flag has different options that can be used in different situation. The commonly used option is `.all`

To test:

- Build/run this branch on both WordPress and Jetpack targets and make sure that the badges and banners are visible in all conditions and behave as expected (visible on WordPress, not visible on Jetpack).
- Also make sure they are not visible if you are not logged in and have a self hosted site that's not jetpack connected
    - Stats
    - Activity log
    - Activity detail
    - Reader
    - Reader Search
    - **Note that this list does not include Notifications, as there is another [PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/19085) that's addressing this case**

## Regression Notes
1. Potential unintended areas of impact
Jetpack powered banners and badges

2. What I did to test those areas of impact (or what existing automated tests I relied on)
test on a few devices

3. What automated tests I added (or what prevented me from doing so)
none
PR submission checklist:
